app/src/main/java/com/github/se/polyfit/viewmodel/meal/OverviewViewModel.kt

```diff
@@ -220,4 +224,29 @@ constructor(
   fun getCaloryGoal(): Long {
     return user.calorieGoal
   }
+
+  fun uriToBitmap(context: Context, uri: Uri): Bitmap? {
+    return context.contentResolver
+        .openInputStream(uri)
+        ?.use { inputStream ->
+          ByteArrayOutputStream().apply { inputStream.copyTo(this) }.toByteArray()
+        }
+        ?.let { BitmapFactory.decodeByteArray(it, 0, it.size) }
+  }
+
+  fun handleSelectedImage(
+      context: Context,
+      uri: Uri?,
+      overviewViewModel: OverviewViewModel,
+  ): String? {
+    if (uri != null) {
+      val imageBitmap =
+          uriToBitmap(context, uri)
+              ?: BitmapFactory.decodeResource(context.resources, R.drawable.picture_example)
+      return overviewViewModel.storeMeal(imageBitmap)
+    } else {
+      Log.d("PhotoPicker", "No media selected")
+      return null
+    }
```

Comment:
```suggestion
     return if (uri != null) {
      val imageBitmap =
        uriToBitmap(context, uri)
          ?: BitmapFactory.decodeResource(context.resources, R.drawable.picture_example)
      overviewViewModel.storeMeal(imageBitmap)
    } else {
      Log.d("PhotoPicker", "No media selected")
      null
    }
```
Simplification that android studio suggests

Reply:
Nice suggestion !